test-framework/mongodb/src/main/java/io/quarkus/test/mongodb/MongoReplicaSetTestResource.java

```diff
@@ -1,110 +1,107 @@
 package io.quarkus.test.mongodb;
 
-import static de.flapdoodle.embed.process.config.process.ProcessOutput.builder;
-
-import java.io.IOException;
-import java.net.UnknownHostException;
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.List;
-import java.util.Map;
-import java.util.concurrent.TimeUnit;
-
-import org.awaitility.Awaitility;
-import org.bson.Document;
-import org.jboss.logging.Logger;
-
 import com.mongodb.ConnectionString;
 import com.mongodb.MongoClientSettings;
 import com.mongodb.client.MongoClient;
 import com.mongodb.client.MongoClients;
 import com.mongodb.client.MongoDatabase;
-
-import de.flapdoodle.embed.mongo.Command;
-import de.flapdoodle.embed.mongo.MongodExecutable;
-import de.flapdoodle.embed.mongo.MongodStarter;
-import de.flapdoodle.embed.mongo.config.*;
+import de.flapdoodle.embed.mongo.commands.MongodArguments;
+import de.flapdoodle.embed.mongo.commands.ServerAddress;
+import de.flapdoodle.embed.mongo.config.Net;
+import de.flapdoodle.embed.mongo.distribution.IFeatureAwareVersion;
 import de.flapdoodle.embed.mongo.distribution.Version;
-import de.flapdoodle.embed.process.config.RuntimeConfig;
-import de.flapdoodle.embed.process.io.Processors;
-import de.flapdoodle.embed.process.runtime.Network;
+import de.flapdoodle.embed.mongo.transitions.Mongod;
+import de.flapdoodle.embed.mongo.transitions.RunningMongodProcess;
+import de.flapdoodle.embed.process.io.ProcessOutput;
+import de.flapdoodle.reverse.TransitionWalker;
+import de.flapdoodle.reverse.transitions.Start;
 import io.quarkus.test.common.QuarkusTestResourceLifecycleManager;
+import org.awaitility.Awaitility;
+import org.bson.Document;
+import org.jboss.logging.Logger;
 
-public class MongoReplicaSetTestResource implements QuarkusTestResourceLifecycleManager {
+import java.io.IOException;
+import java.net.UnknownHostException;
+import java.util.*;
+import java.util.concurrent.TimeUnit;
 
+public class MongoReplicaSetTestResource implements QuarkusTestResourceLifecycleManager {
     private static final Logger LOGGER = Logger.getLogger(MongoReplicaSetTestResource.class);
-    private static List<MongodExecutable> MONGOS = new ArrayList<>();
+    private Integer port;
+    private IFeatureAwareVersion version;
+
+    private List<TransitionWalker.ReachedState<RunningMongodProcess>> startedServers=Arrays.asList();
```

Comment:
There's no point to create a new empty list then replace it with the list of the two servers.
Would be better to keep the original way of instanciating a new arraylist and adding to it.

Reply:
I tried to reduce the chance for a NPE.. but can change it the way you want.